src/system_identity/http_token_retriever.rs

```diff
@@ -0,0 +1,79 @@
+use std::fmt;
+
+use http::Uri;
+
+use crate::{
+    authenticator::HttpAuthenticator,
+    http_client::HttpClient,
+    jwt::signer::{local::LocalPrivateKeySigner, JwtSignerImpl},
+    token::Token,
+    token_retriever::TokenRetrieverWithCache,
+    TokenRetriever, TokenRetrieverError,
+};
+
+use super::{client_input::AuthMethod, l1_token_retriever::L1TokenRetriever};
+
+pub enum HttpTokenRetrieverImpl<'a, C: HttpClient> {
```

Comment:
Nit: I would call it `TokenRetrieverImpl` or `HttpTokenRetriever` (there is no  `HttpTokenRetriever` trait, there is?)

Reply:
I agree, `HttpTokenRetriever` is the most fitting in my view given the type param trait bound. I'll rename!

Reply:
Renamed in https://github.com/newrelic/newrelic-auth-rs/pull/25/commits/8739a25c4cd997613fba6d145a4fe03859954e0d!